ui/components/search/search.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import querystring from 'querystring';
+
+import { redirectQuery } from '../lookup-search';
+
+export default class Search extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { searchTerm: '' };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ searchTerm: e.target.value });
+  }
```

Comment:
Adding state here is a bit of a mixed bag. On one hand, it means that changes to the rest of the page won't affect the state of this input (so the text the user wrote will remain, even if they change the URL by adding/removing a filter), if I understand correctly. On the other hand, it makes the logic harder to follow since we don't have a simple data-in-xml-out situation; for example, do we ever need to clear the state? What do you think we should do?

If we're going to keep state, I'd like to add a test to validate that state gets set onChange.